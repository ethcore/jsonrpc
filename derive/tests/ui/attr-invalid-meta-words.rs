extern crate serde;
extern crate jsonrpc_core;
#[macro_use]
extern crate jsonrpc_derive;

use jsonrpc_core::Result;

#[rpc]
pub trait Rpc {
	/// Returns a protocol version
	#[rpc(name = "protocolVersion", Xmeta)]
	fn protocol_version(&self) -> Result<String>;
}

fn main() {}
