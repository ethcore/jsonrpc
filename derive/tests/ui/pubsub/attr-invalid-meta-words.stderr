error: Invalid attribute parameter(s): 'Xsubscribe'. Expected 'subscribe, unsubscribe'
  --> $DIR/attr-invalid-meta-words.rs:10:2
   |
10 |       /// Hello subscription
   |  _____^
11 | |     #[pubsub(subscription = "hello", Xsubscribe, name = "hello_subscribe", alias("hello_sub"))]
12 | |     fn subscribe(&self, Self::Metadata, typed::Subscriber<String>, u64);
   | |________________________________________________________________________^

error: aborting due to previous error

